src/main/java/org/mitre/synthea/export/CSVExporter.java

```diff
@@ -676,10 +679,20 @@ private String encounter(RandomNumberGenerator rand, String personID,
       s.append(',');
     }
     // CODE
-    Code coding = encounter.codes.get(0);
-    s.append(coding.code).append(',');
-    // DESCRIPTION
-    s.append(clean(coding.display)).append(',');
+    Code coding = null;
+    try {
+      coding = encounter.codes.get(0);
+      // CODE
+      s.append(coding.code).append(',');
+      // DESCRIPTION
+      s.append(clean(coding.display)).append(',');
+    } catch (IndexOutOfBoundsException e) {
+      // Sometimes an encounter may not have a code. In this case, catch the error
+      // and add a bandaid.
+      System.out.println("Missing encounter code in CSV Exporter.");
+      s.append("INVALID_CODE").append(',');
+      s.append(clean("INVALID_CODE")).append(',');
+    }
```

Comment:
I don't totally recall. I remember there was an issue that would rarely be triggered when split_records were enabled and this may be it. I think it had something to do with an encounter getting started in one record and then not getting completed before a new record would begin.